### PR TITLE
added modal for related items and started on outfit

### PR DESCRIPTION
**Pull request by hangyin2020**
added modal for related item list

### CODE REVIEW

File: client/src/components/RelatedItems/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const Overlay = styled.div`
+position: fixed;
```
Comment (zman811) [inline client/src/components/RelatedItems/Modal.jsx:6]: When writing styled components make sure to tab out the styles

File: client/src/components/RelatedItems/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  right:400px;
+  bottom:200px;
+  background-color: rgba(0, 0, 0, 0.5);
+  display: flex;
+  align-items:center;
+  justify-content: center;
+`;
+
+const Body = styled.div`
+padding: 10px;
```
Comment (zman811) [inline client/src/components/RelatedItems/Modal.jsx:30]: same thing here

**Review by zman811 — APPROVED** [review]
Looks good, one thing I would say to change later is you could change the compare button to just be what happens when you click on the picture, but otherwise nice styling!

File: client/src/components/RelatedItems/Modal.jsx (hunk 2 above)
Comment (hangyin2020) [reply client/src/components/RelatedItems/Modal.jsx:30]: thanks for reviewing, will tap those out for sure

**Review by slargman — CHANGES_REQUESTED** [review]
There seems to be some issues with the automated checks not running. I'll merge the latest changes from main in to this branch to see if we can get that resolved before completing the PR.

Other than that overall it looks really good. I suggested some variable naming conventions that might make it easier for you and others to follow your code. I also pointed out what may be some simpler ways of doing certain operations. None of those changes are necessary to make. It's more a matter of style and potentially cleaner code than functionality.

However, would you mind only adding one component to the top-level App component? It will help maintain modularization if we each only supply one component instead of several subcomponents. There are also a few places where keys are missing for lists and it'd be good to make sure there are keys there to avoid bugs.

**Comment by slargman** [comment]
Looks like the build checks are good!

File: client/src/components/App/App.jsx [hunk 3]
```diff
@@ -13,7 +14,8 @@ function App() {
     <Container>
       <Overview productId={11}/>
       <Reviews />
-      <RelatedItemsList />
+      <RelatedItemsList currentItemId={40346}/>
+      <OutfitItemsList />
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/App/App.jsx:18]: Thanks for reviewing, those reviews make a lot of sense. The change is made.

File: client/src/components/RelatedItems/Modal.jsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  height: 300px
+  margin: 10px;
+  padding: 10px;
+  color: #eee;
+  border-top: 1px solid #eee;
+  border-bottom: 1px solid #eee;
+`;
+
+function Modal({
+  showModal, onClose, currentChars, currentItemVals, currentRelatedVals,
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/Modal.jsx:51]: The change is made.

File: client/src/components/RelatedItems/Modal.jsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  console.log('currentItemVals: ', currentItemVals);
+  console.log('currentChars: ', currentChars);
+  console.log('currentRelatedVals: ', currentRelatedVals);
+
+  return (
+    <Overlay onClick={onClose}>
+      <Content>
+        <BodyWrapper>
+          {currentItemVals.map((item) => (
+            <Body>
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/Modal.jsx:65]: The change is made.

File: client/src/components/RelatedItems/Modal.jsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+              {item}
+            </Body>
+          ))}
+        </BodyWrapper>
+        <BodyWrapper>
+          {currentChars.map((item) => (
+            <Body>
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/Modal.jsx:72]: The change is made.

File: client/src/components/RelatedItems/Modal.jsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+              {item}
+            </Body>
+          ))}
+        </BodyWrapper>
+        <BodyWrapper>
+          {currentRelatedVals.map((item) => (
+            <Body>
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/Modal.jsx:79]: The change is made.

File: client/src/components/RelatedItems/OutfitItemsList.jsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    <OutfitItemsEntry
+      key={item.id}
+      item={item}
+    />
+  ));
+
+  return (
+    <div>
+      <div>YOUR OUTFIT</div>
+      <Carousel items={OutfitItemsEntries} size={2} />
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/OutfitItemsList.jsx:23]: sounds good

File: client/src/components/RelatedItems/RelatedItemsEntry.jsx [hunk 9]
```diff
@@ -1,34 +1,60 @@
...
+  background-color: #555;
+  color: white;
+  font-size: 15px;
+  padding: 5px 5px;
+  border: none;
+  cursor: pointer;
+  border-radius: 5px;
+`;
+
+function RelatedItemsEntry({ img, item, onOpen, onCompare }) {
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/RelatedItemsEntry.jsx:28]: The change is made.

File: client/src/components/RelatedItems/RelatedItemsList.jsx [hunk 10]
```diff
@@ -1,25 +1,155 @@
...
+
+  const fetchFeatures = async (relatedId) => {
+    const relatedFeatures = [];
+    const currentFeatures = [];
+    await (
+      axios.get(`/products/${relatedId}`)
+        .then((res) => (res.data.features.forEach((feature) => {
+          relatedFeatures.push(feature);
+        })))
+        // [{f1: value}, {f2: value}...]
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/RelatedItemsList.jsx:60]: man I was laughing after I saw this comment...why did I do that...

File: client/src/components/RelatedItems/RelatedItemsList.jsx [hunk 11]
```diff
@@ -1,25 +1,155 @@
...
+        .then((res) => (res.data.features.forEach((feature) => {
+          currentFeatures.push(feature);
+        })))
+        // [{f1: value}, {f2: value}...]
+    );
+    console.log('relatedFeatures: ', relatedFeatures);
+    console.log('currentFeatures: ', currentFeatures);
+    // create allKeys(array), push object1.keys and object2.keys without duplicate
+    const relatedKeys = [];
+    relatedFeatures.forEach((item) => (relatedKeys.push(item.feature)));
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/RelatedItemsList.jsx:73]: this is changed. now I know the importance of underscore functions

File: client/src/components/RelatedItems/RelatedItemsList.jsx [hunk 12]
```diff
@@ -1,25 +1,155 @@
...
+    console.log('relatedKeys: ', relatedKeys);
+
+    const currentKeys = [];
+    currentFeatures.forEach((item) => (currentKeys.push(item.feature)));
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/RelatedItemsList.jsx:77]: this is changed.

File: client/src/components/RelatedItems/RelatedItemsList.jsx [hunk 13]
```diff
@@ -1,25 +1,155 @@
...
+    console.log('currentKeys : ', currentKeys);
+
+    const allFeatures = relatedKeys.slice();
+    currentKeys.forEach((key) => {
+      if (allFeatures.indexOf(key) < 0) {
```
[the remark this answers is not in the document]
Comment (hangyin2020) [reply client/src/components/RelatedItems/RelatedItemsList.jsx:82]: this is changed.

**Review by slargman — APPROVED** [review]
Changes look great! Thank you for making them.

**Pull request merged**